neovim/plugin/script_host.py

```diff
@@ -44,6 +44,14 @@ def __init__(self, nvim):
         exec('import sys', self.module.__dict__)
         self.legacy_vim = LegacyVim.from_nvim(nvim)
         sys.modules['vim'] = self.legacy_vim
+        # Since the 'DirChanged' autocmd is established asynchronously, import
+        # os manually already, to prevent a potential race condition
+        exec('import os', self.module.__dict__)
```

Comment:
Why is this needed? The autocmd handler method is executed in the context of `script_host.py` which already has the import.

Reply:
I tried to work-around the race you described, by importing os before, so that Vim can see the current working directory without creating a race.

Reply:
But the race has nothing to do with `import os`. Quick fix would rather be to do 
`nvim.command('call rpcrequest({}, "python_chdir", {"cwd": getcwd()}', async=True)` after registering the autocmd

Reply:
The quickfix doesn't seem to work, at least for me, it results in a python error from my plugins.
<img width="592" alt="screen shot 2018-01-24 at 10 11 03" src="https://user-images.githubusercontent.com/396475/35323780-07b465aa-00ef-11e8-88ee-8585d6ba8381.png">